great_expectations/profile/user_configurable_profiler.py

```diff
@@ -75,42 +77,40 @@ def __init__(
+        Args:
+            profile_dataset: A Great Expectations Dataset or Validator object.
+            excluded_expectations: A  list of Expectations to omit from the suite.
+            ignored_columns: A list of columns for which you would like to NOT create Expectations.
+            not_null_only: By default, each column is evaluated for nullity. If the
+                column values contain fewer than 50% null values, then the Profiler will add
+                `expect_column_values_to_not_be_null`; if greater than 50% it will add
+                `expect_column_values_to_be_null`. If `not_null_only` is set to `True`, the Profiler will add a
+                `not_null` Expectation irrespective of the percent nullity (and therefore will not add
+                `expect_column_values_to_be_null`).
+            primary_or_compound_key: A list containing one or more columns which are a dataset's primary or
+                compound key. This will create an `expect_column_values_to_be_unique` or
+                `expect_compound_columns_to_be_unique` Expectation. This will occur even if one or more of the
+                `primary_or_compound_key` columns are specified in `ignored_columns`.
+            semantic_types_dict: A dict where the keys are available semantic types
+                (see `profiler.base.ProfilerSemanticTypes`) and the values are lists of columns for which you
+                would like to create semantic-type-specific Expectations e.g.:
+                `"semantic_types": { "value_set": ["state","country"], "numeric":["age", "amount_due"]}`
+            table_expectations_only: If `True`, this will only create the two table level
+                Expectations available to this Profiler (`expect_table_columns_to_match_ordered_list` and
+                `expect_table_row_count_to_be_between`). If `primary_or_compound_key` is specified, it will
+                create a uniqueness Expectation for that column as well.
+            value_set_threshold: Must be one of: `"none"`, `"one"`, `"two"`, `"very_few"`, `"few"`, `"many"`, `"very_many"`, `"unique"`.
+                When the Profiler runs without a `semantic_types_dict`, each column is profiled for cardinality. This threshold determines the
+                greatest cardinality for which to add `expect_column_values_to_be_in_set`. For example, if
+                `value_set_threshold` is set to `"unique"`, it will add a `value_set` Expectation for every included
+                column. If set to `"few"`, it will add a `value_set` Expectation for columns whose cardinality is
+                one of `"one"`, `"two"`, `"very_few"`, or `"few"`. For the purposes of
+                comparing whether two tables are identical, it might make the most sense to set this to `"unique"`.
+
+        Raises:
+            ValueError: If an invalid `primary_or_compound_key` is provided.
```

Comment:
These should move up to the class docstring instead of under the `__init__()` constructor